File: Session4/Java/ReverseWordsInString.java
https://leetcode.com/problems/reverse-words-in-a-string-iii/

class Solution {
    
    void reverse_str(char[] s, int start, int end)
    {
        while(start<end)
        {
            char temp = s[start];
            s[start] = s[end];
            s[end] = temp;
            start++;
            end--;
        }
    }
    
    public String reverseWords(String s) {
        
        char[] c = s.toCharArray();
        int i=0;
        for(int j=0; j<c.length; j++)
        {
            if(c[j]==' ')
            {
                reverse_str(c, i, j-1);
                i = j+1;
            }
        }
        
        reverse_str(c, i, c.length-1);
        
        return new String(c);
        
    }
}

File: Session4/C++/ReverseWordsInString.cpp
https://leetcode.com/problems/reverse-words-in-a-string-iii/

class Solution {
    
    void reverse_str(string &s, int start, int end)
    {
        while(start<end)
        {
            swap(s[start], s[end]);
            start++;
            end--;
        }
    }
    
public:
    string reverseWords(string s) {
        
        int i=0;
        for(int j=0; j<s.length(); j++)
        {
            if(s[j]==' ')
            {
                reverse_str(s, i, j-1);
                i = j+1;
            }
        }
        
        reverse_str(s, i, s.size()-1);
        
        return s;
        
    }
};